functions/consumer/tcp-consumer/src/main/java/org/springframework/cloud/fn/consumer/tcp/TcpConsumerConfiguration.java

```diff
@@ -38,28 +37,28 @@
  *
  * @author Gary Russell
  * @author Christian Tzolov
+ * @author Chris Bono
  */
-@Configuration
+@Configuration(proxyBeanMethods = false)
 @EnableConfigurationProperties({TcpConsumerProperties.class, TcpConnectionFactoryProperties.class})
 public class TcpConsumerConfiguration {
 
-	@Autowired
 	private TcpConsumerProperties properties;
-
-	@Autowired
 	private TcpConnectionFactoryProperties tcpConnectionProperties;
 
-	@Qualifier("tcpSinkConnectionFactory")
-	@Autowired
-	private AbstractConnectionFactory connectionFactory;
+	public TcpConsumerConfiguration(TcpConsumerProperties properties,
+									TcpConnectionFactoryProperties tcpConnectionProperties) {
+		this.properties = properties;
+		this.tcpConnectionProperties = tcpConnectionProperties;
+	}
 
 	@Bean
-	public Consumer<Message<?>> tcpConsumer() {
-		return handler()::handleMessage;
```

Comment:
Again, the config trying to use the thing that it creates via field injection. I moved this to constructor injection and bean method param injection.